static/js/containers/UpsellCardApp.js

```diff
@@ -0,0 +1,138 @@
+// @flow
+/* global SETTINGS:false */
+import React, { Fragment } from "react"
+import { createStructuredSelector } from "reselect"
+import { pathOr } from "ramda"
+import { compose } from "redux"
+import { connect } from "react-redux"
+import { connectRequest } from "redux-query"
+import { Badge } from "reactstrap"
+
+import Loader from "../components/Loader"
+import { routes } from "../lib/urls"
+import { EnrollmentFlaggedCourseRun } from "../flow/courseTypes"
+import {
+  courseRunsSelector,
+  courseRunsQuery,
+  courseRunsQueryKey
+} from "../lib/queries/courseRuns"
+
+import { isWithinEnrollmentPeriod } from "../lib/courseApi"
+
+import { getCookie } from "../lib/api"
+import type { User } from "../flow/authTypes"
+import users, { currentUserSelector } from "../lib/queries/users"
+
+type Props = {
+  courseId: string,
+  isLoading: ?boolean,
+  courseRuns: ?Array<EnrollmentFlaggedCourseRun>,
+  status: ?number,
+  upgradeEnrollmentDialogVisibility: boolean,
+  addProductToBasket: (user: number, productId: number) => Promise<any>,
+  currentUser: User
+}
+type ProductDetailState = {
+  upgradeEnrollmentDialogVisibility: boolean
+}
+
+export class UpsellCardApp extends React.Component<Props, ProductDetailState> {
+  state = {
+    upgradeEnrollmentDialogVisibility: false
+  }
+
+  toggleUpgradeDialogVisibility = () => {
+    const { upgradeEnrollmentDialogVisibility } = this.state
+    this.setState({
+      upgradeEnrollmentDialogVisibility: !upgradeEnrollmentDialogVisibility
+    })
+  }
+
+  renderUpgradeEnrollmentDialog(run: EnrollmentFlaggedCourseRun) {
+    console.log(run)
+    const { courseRuns } = this.props
+    const product = run.products && !run.is_verified ? run.products[0] : null
+
+    return product ? (
+      <div className="card">
+        <div className="row d-flex upsell-header">
+          <div className="flex-grow-1 align-self-end">
+            <Badge color="danger">Enrolled in free course</Badge>
+            <h2>Get Certificate</h2>
+          </div>
+          <div className="text-right align-self-end">
+            <h2>${product.price}</h2>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-12">
+            <p>
+              <strong>
+                You are enrolled in the free version of this course.
+              </strong>
+            </p>
+            <p>
+              <span className="font-weight-bold">Get a certificate today</span>{" "}
+              and lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
+              do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
```

Comment:
This copy needs to be updated to match the text quoted in the issue here: https://github.com/mitodl/mitxonline/issues/440

You can disregard the text in the mockups since those haven't been updated.